### PR TITLE
Pulled Vault core into a separate submodule

### LINKED ISSUE #98: Pull server and load balancer config into a submodule
For users that have more strict separation of duties or those that use Shared VPC, creating a network and firewall rules will not work. Instead of having many switches for what resources to enable or disable, we should have a seperate submodule specifically for the LB and Vault servers so networking teams can add the other pieces ahead of time and potentially in other projects.

### PR DESCRIPTION
**Pull request by onetwopunch**
Fixes #98 

# Breaking change, major release.

This refactor pulls out the server, tls and load balancer resources into their own module called `cluster` so that larger enterprises that have separate teams to configure things like buckets, firewalls, networks, and service accounts can still use the core Vault logic.

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -18,6 +18,6 @@ terraform {
   required_version = ">= 0.12.6"
 
   required_providers {
-    google = "~> 3.15"
+    google = "~> 3.30.0"
```
Comment (bharathkkb) [inline versions.tf:21]: Are we depending on some `3.30` feature? I maybe wrong but I think `~> 3.30.0` will lock the module to `>=3.30.0, <3.31.0`

File: test/setup/main.tf [hunk 2]
```diff
@@ -14,20 +14,70 @@
...
+}
 
-  activate_apis = [
-    "cloudresourcemanager.googleapis.com",
-    "serviceusage.googleapis.com",
-    "compute.googleapis.com",
-    "iam.googleapis.com",
+module "svpc" {
+  source          = "terraform-google-modules/network/google"
+  project_id      = module.project_ci.project_id
```
Comment (bharathkkb) [inline test/setup/main.tf:52]: pin modules here and throughout

File: test/setup/make_source.sh [hunk 3]
```diff
@@ -19,6 +19,9 @@ echo "#!/usr/bin/env bash" > ../source.sh
 project_id=$(terraform output project_id)
 echo "export TF_VAR_project_id='$project_id'" >> ../source.sh
 
+svc_project_id=$(terraform output service_project_id)
+echo "export TF_VAR_service_project_id='$svc_project_id'" >> ../source.sh
```
Comment (bharathkkb) [inline test/setup/make_source.sh:23]: @morgante I believe we can remove this now? AFAIK we use [this](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/build/developer-tools/build/scripts/export_tf_outputs.py) now instead automatically surfaces these.

Comment (onetwopunch) [reply test/setup/make_source.sh]: Thanks for pointing this out! Removed the file and it works perfectly :)

**Review by AmandaG133 — APPROVED** [review]

**Pull request merged**